app/src/main/java/com/example/login/ui/wifi/WifiFragment.kt

```diff
@@ -0,0 +1,190 @@
+                    name,
+                    requestCode
+                )
+                else -> ActivityCompat.requestPermissions(
+                    context as Activity,
+                    arrayOf(permission),
+                    requestCode
+                )
+            }
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        fun innerCheck(name: String) {
+            if (grantResults.isEmpty() || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                Toast.makeText(context, name + " " + getString(R.string.refused), Toast.LENGTH_SHORT).show()
+            } else {
+                Toast.makeText(context, name + " " + getString(R.string.granted1), Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        when (requestCode) {
+            FINE_LOCATION_RQ -> innerCheck(location)
+        }
+    }
+
+    private fun showDialog(permission: String, name: String, requestCode: Int) {
+        val builder = AlertDialog.Builder(context)
+        builder.apply {
+            setMessage(getString(R.string.perm_access) + name)
+            setTitle(getString(R.string.granted1))
+            setPositiveButton("OK") { dialog, which ->
+                ActivityCompat.requestPermissions(
+                    context as Activity,
+                    arrayOf(permission),
+                    requestCode
+                )
+            }
+        }
+        val dialog = builder.create()
+        dialog.show()
+    }
+
+
+    private fun wifiAnimation() {
+        val animAlpha = AnimationUtils.loadAnimation(context, R.anim.button_anim);
+        binding.tvWifiScan.startAnimation(animAlpha);
+    }
+
+
+    private fun createWifi(wifiList: ArrayList<WifiModel>, context: Context) {
+        wifiList.forEach { wifiObject ->
+            val oneWifiItem = layoutInflater.inflate(R.layout.wifi_item, null)
+            val wifiName = oneWifiItem.findViewById<TextView>(R.id.tvWifiName)
+            val wifiPower = oneWifiItem.findViewById<TextView>(R.id.tvWifiPower)
+            myConstrain.addView(oneWifiItem)
+
+            wifiName.text = wifiObject.name
+            val params = oneWifiItem.layoutParams as ConstraintLayout.LayoutParams
+            params.circleConstraint = R.id.tvWifiScan
+            params.circleAngle = angle
+
+            when {
+                wifiObject.power > -50 -> {
+                    wifiPower.text = getString(R.string.excellent)
+                    params.circleRadius = 150
+                }
+                wifiObject.power in -50 downTo -64 -> {
+                    wifiPower.text = getString(R.string.good)
+                    params.circleRadius = 250
+                }
+                wifiObject.power in -65 downTo -74 -> {
+                    wifiPower.text = getString(R.string.fair)
+                    params.circleRadius = 500
```

Comment:
why circleradius is fixed ? rewrite render based on screen width and height

Reply:
done